### DISCUSSION TITLE
Customize int_mat in Volterra_IDE.py

### DISCUSSION THREAD

**Discussion by Bensayah** [lululxvi/deepxde#826] [Q&A]
Hello
How to customize int_mat in Volterra_IDE.py

Many thanks

**Comment by lululxvi** [comment]
What do you mean?

**Reply by Bensayah** [reply]
I want to use a quadratic approximation of the integral of the form \int_0^t s**(p-1)(t-s)**(q-1)f(s)ds where weights involve the integral \int_0^t s**(p-1)(t-s)**(q-1) contrary to legender.lelgauss method used in the source. Then we can approximate problems using the Kernel  K(t,s)= s**(p-1)(t-s)**(q-1)  in particular epidemic problems described by fractional integro-differential systems.

**Reply by lululxvi** [reply]
Then you can modify the source code.

**Reply by Bensayah** [reply]
Hi lulu thanks for the replay
My next step is extend the model to a system of integral equations which is supported by Volterra_IDE program. But when enlarging the time dependent domain I obtained a constant solutions of course their values are the correspondent initial values. What to do to overcome this problem?

**Reply by lululxvi** [reply]
I am not sure what the problem is. If it is due to large time domain, then there are some papers on this.

**Comment by Bensayah** [comment]
Yes this is the first part of the problem the second one is how to avoid
trivial solutions.

Le jeu. 11 août 2022 à 03:00, Lu Lu ***@***.***> a écrit :

> I am not sure what the problem is. If it is due to large time domain, then
> there are some papers on this.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/lululxvi/deepxde/discussions/826#discussioncomment-3371097>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AII5BBG523P2WCOTKDZNA6DVYRNFRANCNFSM544VBIMQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
A. Bensayah
Department of mathematics
Ouargla university, Algeria
Phone:+213 0697808010
ORCID ID: https://orcid.org/0000-0001-7937-4694
Google site: https://sites.google.com/view/bensayah-abdallah/home
Google Scholar: https://scholar.google.com/citations?hl=en&user=UJhhQD4AAAAJ

Researchgate address: https://www.researchgate.net/profile/Bensayah_Abdallah
Academia address: https://ouargla
<https://www.google.com/url?q=https%3A%2F%2Fouargla&sa=D&sntz=1&usg=AFQjCNFfEHjsCv_xBA1OrsySAaUaZJUl8w>
-univ.dz.academia.edu/AbdallahBensayah
Github address: https://github.com/Bensayah
Jovian address: https://jovian.ai/bensayahabd
Kaggle address: https://www.kaggle.com/abdallahbensayah

**Comment by lululxvi** [comment]
See FAQ "Q: I failed to train the network or get the right solution, e.g., large training loss, unbalanced losses."

**Reply by Bensayah** [reply]
Hi Lulu
I have fixed what to do to make ide.py handle the case of:
 "*Systems of integral equations with fractal-fractional kernel    s^(p-1)*(t-s)^(q-1) "
Contributions: System for large interval time - supports singular kernel.
The graphs coincide with those obtained by another numerical method namely
Adams-Bashforth method.
Is it possible for corporations to publish these results?
Many thanks

**Reply by lululxvi** [reply]
Sounds good. Yes, you can publish the results.

**Reply by Bensayah** [reply]
Hi Dr. LuLu
In order to complete the draft. The program works very well to equations and systems with single entrie $t$. I try to test the program on equations having unknown functions of the form  $u(x,t)$. Plz, how to modify ide.py  to handle problems with entries $(x,t)$? The issue #198 does not work for me.
Many thanks in advance.

**Reply by lululxvi** [reply]
Once you understand the code, then you should be able to modify. I have no experience with this.
